README.md

```diff
@@ -1 +1,79 @@
-# skill-tree-backend
\ No newline at end of file
+
+# Skill Tree Backend
+
+## Steps to Run the Service Locally
+
+### Required Tools
+
+- [Maven](https://mvnrepository.com/)
+- Java
+- Integrated Development Environment (IDE)
+- Docker Desktop
+- MySQL Docker Image
+
+### Steps to Connect the Service to MySQL Running in Docker
+
+Refer to this [link](https://find10archived.medium.com/how-to-connect-a-mysql-docker-container-with-a-local-spring-boot-application-9366707dce0d) for detailed steps.
+
+1. Create a database user with full access for table creation, deletion, and modification (for development only in local MySQL).
+2. After creating the database user, enter the credentials in the Skill Tree Spring Boot application.
+3. Attempt to connect to the database using the URL "jdbc:mysql://${MYSQL_HOST:localhost}:3306/${DB_NAME}".
+4. Create a simple API to test the database connection and data retrieval. For example, create a `/test` route in `skill-tree/src/main/java/com/RDS/skilltree/User/UserController.java`.
+    ```java
+    package com.RDS.skilltree.User;
+
+    import org.springframework.web.bind.annotation.GetMapping;
+    import org.springframework.web.bind.annotation.RestController;
+
+    @RestController
+    public class UserController {
+        @GetMapping("/test")
+        public void test(){
+            System.out.println("test123");
+        }
+    }
+    ```
+
+### Steps to Generate a JWT Token (Similar to [Website Backend Repo](https://github.com/Real-Dev-Squad/website-backend/issues))
+
+1. Ensure the website-backend is running locally.
+2. Generate Public and Private keys using openssl with the following commands:
+   - `openssl genrsa -out ./private.key 4096`
+   - `openssl rsa -in private.key -pubout -outform PEM -out public.key`
+3. Obtain a user token (private and public key stored in `local.js` file).
+4. After calling `localhost:3000/auth/github/login`, the backend will generate the JWT token.
+5. Validate the token using [jwt.io](https://jwt.io/) by entering the public and private keys stored in the website backend.
+6. Use the public key in the Skill Tree repo to decrypt the JWT token passed for authentication.
+
+## Steps for Creating the Database
+
+1. `create database skilltree;`
+2. `show databases;`
+3. `create user 'testuser' identified by 'testpassword';` (Username: testuser, Password: testpassword)
+4. `grant all on skilltree.* to testuser;`
+
```

Comment:
## Connect mysql workbench to running mysql inside docker
1. `docker exec -it rds-db-1 bin/bash`
2. bash-4.4# `mysql -u root -p -A`

By default after deployment MySQL has following connection restrictions:
```
mysql> select host, user from mysql.user;
+-----------+---------------+
| host      | user          |
+-----------+---------------+
| localhost | healthchecker |
| localhost | mysql.session |
| localhost | mysql.sys     |
| localhost | root          |
+-----------+---------------+
4 rows in set (0.00 sec)
```
3. update mysql.user set host='%' where user='root';
4. Quit the mysql client.
5. Restart the docker container

Now you can connect to the mysql running in the docker container, also to connect to it on the terminal type `mysql -utestuser -p -P3306 -h127.0.0.1`

Reply:
Yes we will have to expose the docker container on port, for external applications to connect.

Reply:
The simpler way is to connect the mysql using adminer as we did @heyrandhir you can try mentioning that too, if someone doesn't has MySql workbench already installed, that we can do with docker container itself need not install anything extra.